src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected readonly PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            if (pyObj == null)
+                throw new PythonException();
+            pyObject = pyObj;
+        }
+
+        private void propagateIterationException()
+        {
+            var err = Runtime.PyErr_Occurred();
+            if (err == null) return;
```

Comment:
`PyErr_Occurred` returns `IntPtr`. You can't compare it to `null`, you have to compare it to `IntPtr.Zero`.